cmd/translate/main_grpc_integration_test.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	pb "go.expect.digital/translate/pkg/server/translate/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+)
+
+func TestMain(m *testing.M) {
+	go main()
+
+	// Needed to prevent panic, when service is not ready to receive requests
+	time.Sleep(time.Millisecond * 50)
```

Comment:
Do your best to avoid using **time.Sleep()**:
- it slows down tests
- it is not deterministic

One of the simplest ways to resolve it is by checking if the port is open https://stackoverflow.com/questions/56336168/golang-check-tcp-port-open .